### PR TITLE
ticdc(sink) move pulsar sink-uri configs to sink of replica config

### PR DESCRIPTION
**Pull request by yumchina**
### What problem does this PR solve?

move pulsar sink-uri configs to sink of replica config
Issue Number: ref https://github.com/pingcap/tiflow/issues/9413

#### Tests 

 - Unit test
 

##### Will it cause performance regression or break compatibility?

No

##### Do you need to update user documentation, design documentation or monitoring documentation?

No

### Release note 
```release-note
None
```

### CODE REVIEW

File: pkg/config/sink.go [hunk 1]
```diff
@@ -331,6 +333,74 @@ type KafkaConfig struct {
...
+func (p *PulsarCompressionType) Value() pulsar.CompressionType {
+	if p == nil {
+		return 0
+	}
+	switch strings.ToLower(string(*p)) {
+	case "lz4":
+		return pulsar.LZ4
+	case "zlib":
+		return pulsar.ZLib
+	case "zstd":
```
Comment (nongfushanquan) [inline pkg/config/sink.go:349]: whether is the  snappy type supported ?

Comment (yumchina) [reply pkg/config/sink.go:349]: Pulsar client doesn't support snappy type

**Review by nongfushanquan — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**